engines/router/missionctl/handlers/batch_http_handler_test.go

```diff
@@ -0,0 +1,171 @@
+package handlers
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+	"time"
+
+	_ "github.com/gojek/turing/engines/experiment/runner/nop"
+	"github.com/gojek/turing/engines/router/missionctl"
+	"github.com/gojek/turing/engines/router/missionctl/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewBatchHTTPHandler(t *testing.T) {
+	missionCtl := createGenericMissionControl(t)
+	batchHTTPHandler := NewBatchHTTPHandler(missionCtl)
+	assert.NotNil(t, batchHTTPHandler)
+
+	//Create test routes endpoint. Route will write request body as response
+	testRouterHandler := http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		requestBody, err := ioutil.ReadAll(request.Body)
+		assert.NoError(t, err)
+		_, err = writer.Write(requestBody)
+		assert.NoError(t, err)
+	})
+	server := createRouteServer(t, testRouterHandler)
+	server.Start()
+	defer server.Close()
+
+	tests := map[string]struct {
+		payload              string
+		expectedResponseBody string
+		expectedStatusCode   int
+		expectedContentType  string
+	}{
+		"ok request": {
+			payload: `[
+							{ "request1": "value1" },
+							{ "request2": "value2" }
+					  ]`,
+			expectedResponseBody: `[
+										{ "code": 200, 
+										  "data": {"request1": "value1"}
+										},
+										{ "code": 200, 
+										  "data": {"request2": "value2"}
+										}
+								  ]`,
+			expectedStatusCode:  200,
+			expectedContentType: "application/json",
+		},
+		"invalid json": {
+			payload:              `[{ : }]`,
+			expectedResponseBody: "Invalid json request\n",
+			expectedStatusCode:   400,
+			expectedContentType:  "text/plain; charset=utf-8",
+		},
+		"invalid json request": {
+			payload:              `{"key": "value1"}`,
+			expectedResponseBody: "Invalid json request\n",
+			expectedStatusCode:   400,
+			expectedContentType:  "text/plain; charset=utf-8",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			req := httptest.NewRequest(http.MethodPost, "/v1/batch_predict", bytes.NewBuffer([]byte(test.payload)))
+			w := httptest.NewRecorder()
+			batchHTTPHandler.ServeHTTP(w, req)
+			res := w.Result()
+			defer res.Body.Close()
+			result, _ := ioutil.ReadAll(res.Body)
+
+			assert.Equal(t, test.expectedStatusCode, res.StatusCode)
+			assert.Equal(t, test.expectedContentType, res.Header.Get("Content-Type"))
+			if test.expectedContentType == "application/json" {
+				assert.JSONEq(t, test.expectedResponseBody, string(result))
+			} else {
+				assert.Equal(t, test.expectedResponseBody, string(result))
+			}
+		})
+	}
+}
+
+func TestNewBatchHTTPHandlerBadRoute(t *testing.T) {
```

Comment:
Can you mergte `TestNewBatchHTTPHandlerBadRoute`  with `TestNewBatchHTTPHandler` by adding `missionCtl ` as a parameter of the test suite:
```go
tests := map[string]struct {
		payload              string
		expectedResponseBody string
		expectedStatusCode   int
		expectedContentType  string
                missionCtl: *missionctl.MissionControl
	}
```
and then dynamically constructing HTTP handler inside the test Run:
```go
for name, test := range tests {
	t.Run(name, func(t *testing.T) {
		batchHTTPHandler := NewBatchHTTPHandler(missionCtl)
		assert.NotNil(t, batchHTTPHandler)
		// ...
	})
}
```